Locator address overriding

mmusselman-crl opened the pull request:
This adds the ability to specify a custom locator address for each endpoint defined in the config. This is useful in niche scenarios where the interface address generated by Zenoh for use in the advertised locator is inconsistent with an external NAT IP address.

For example, one could define an endpoint like so:
```json
{
  "endpoints": ["tls/0.0.0.0:1234#loc_addr=192.168.0.200"]
}
```

This can be combined with other connection parameters like `iface` by separating options with semicolons, like `#loc_addr=192.168.0.200;iface=eth0`.

Feel free to suggest any stylistic changes that you prefer (variable names, adding/removing debug printouts, etc.).

diogomatsubara commented:
Hi @mmusselman-crl, to accept contributions we need contributors to have signed the Eclipse ECA. The instructions can be found here: https://api.eclipse.org/git/eca/status/gh/eclipse-zenoh/zenoh/2039. Thanks!

mmusselman-crl commented:
I've signed the ECA now.

mmusselman-crl commented:
I've updated the PR to include a test that should hopefully satisfy the code coverage check. I was able to get it to pass locally, but I'm having trouble getting some other tests to pass. Those tests seem to be mostly unrelated to my PR, though, so it might just be due to differences between my computer and the CI environment.

I also added a couple of lines to the `.json5` files, since they seem to be the only places where endpoint arguments like `#iface` are documented.
